src/sentry/models/projectkey.py

```diff
@@ -286,7 +286,18 @@ def write_relocation_import(
         if old_pk is None:
             return None
 
-        (key, _) = self.__class__.objects.get_or_create(
+        # If there is a key collision, generate new keys.
+        #
+        # TODO(getsentry/team-ospo#190): Is this the right behavior? Another option is to just
+        # update the SQL schema to no longer require uniqueness.
+        matching_public_key = self.__class__.objects.filter(public_key=self.public_key).first()
+        if not self.public_key or matching_public_key:
+            self.public_key = self.generate_api_key()
```

Comment:
Assuming a key collision is super uncommon. However, in the case there is a matching secret/public key, will it be sufficient enough to try regenerating this once? Does this mean it could technically be the same key even after regenerating?

Reply:
This is a 16-byte value. Someone else did the collision likelihood math for us here: https://stackoverflow.com/a/18970981

I'm willing to take a 2^-128 shot that we won't get duplicates in production. We're more likely to see breakages due to colliding hashmap keys in any of the thousands of maps (or `dict`s) we use than we are to see one here.

Reply:
Thank you internet stranger haha